src/adr/0040-quality-levels-definitions.md

```diff
@@ -0,0 +1,62 @@
+# 40. Quality levels definition
+
+Date: 2020-06-21
+
+## Status
+
+Accepted (lazy consensus)
+
+## Context
+
+We hereby define as an artifact compiled artifact that external people consumes. This includes:
+
+ - libraries
+ - Mail servers
+ - Extensions for James Mail Servers
+ - Command line tools
+
+We designate as a feature an optional, opt-in behaviour of a James server that can be configured by 
+user willing to rely on it.
+
+James as a project delivers several artifacts, and features. In order for project users to better
+understand the underlying quality of the artifact they use, as well as the level of risk associated,
+we need to better define some quality levels.
```

Comment:
The document defines "quality levels" without defining "quality".

I think that "level" is obvious enough that it doesn't require any further definition, but "quality" may be interpreted differently by different people.

Reply:
> The document defines "quality levels" without defining "quality".

At one point it becomes hard for me to betterdefines such a generic term. I believes the levels definition disambiguate clearly this.

Reply:
Writing is really hard, I agree. The hardest part is writing something "objective" that everybody understands the same way, especially in a project like this with people from different countries, backgrounds, cultures, education, generations... We'll never have a completely common understanding, so the trick is to decide the right point at which we should draw a boundary given the time and resources we have available. We will never completely agree on where that boundary should be. :)

Reply:
Actually this document, and the "mature" quality level dis-ambiguify what we mean by quality level, no?

Reply:
Are you saying that "mature" implies "high quality"? In the definitions I proposed, a component could be mature, but unreliable.

I do agree that it's not easy figuring out how all the pieces fit together. Quality really depends on the objective. There is not "absolute". That is why context is so important.